File: go/src/prime_factors/prime_factors.go
package prime_factors

func Factors(num int) []int {
        if num == 1 { return []int{} }
	for i := 2; i < num; i++ {
		if num % i == 0 {
			return append([]int{i}, Factors(num/i)...)
		}
	}
        return []int{num}
}

File: scala/src/main/scala/PrimeFactors.scala
package com.crebma.primeFactors

object PrimeFactors {

  def factors(num: Int) : List[Int] = num match {
    case 1 => List[Int]()
    case 2 => List(num)
    case _ =>
      var d = 2;
      while (num%d != 0) {
        d = d + 1
      }
      if (num == d) {
        List(num);
      }
      else {
        List.concat(List(d), factors(num/d));
      }
  }

}
